pkg/apis/allocation/v1/gameserverallocation.go

```diff
@@ -174,6 +212,65 @@ func (s *GameServerSelector) Matches(gs *agonesv1.GameServer) bool {
 		}
 	}
 
+	if runtime.FeatureEnabled(runtime.FeatureCountsAndLists) {
+		return s.matchCountsAndLists(gs)
+	}
+
+	return true
+}
+
+func (s *GameServerSelector) matchCountsAndLists(gs *agonesv1.GameServer) bool {
+	// Match Counter Selectors
+	if s.Counters != nil && gs.Status.Counters != nil {
```

Comment:
Does GameServer has `Status.Counters` default to an empty map when it doesn't have any counter? If not, the logic here will be problematic since it will return true for such a GameServer even the GameServerSelector is asking for some counters.

Reply:
Yes, it should default to nil, good catch on the logic.